### PR TITLE
Stop using deprecated `keyIds` param on /key/v2/server

### LINKED ISSUE #14523: Synapse uses deprecated `keyId` path param on `/_matrix/key/v2/server`
The spec has deprecated the `keyId` parameter on [`GET /_matrix/key/v2/server`](https://spec.matrix.org/v1.5/server-server-api/#get_matrixkeyv2serverkeyid) since [forever](https://github.com/matrix-org/matrix-doc/pull/1423).

(The reason given for that deprecation was that it is *ignored* on the server side.)

### PR DESCRIPTION
**Pull request by richvdh**
Fixes #14523.

### CODE REVIEW

File: synapse/crypto/keyring.py [hunk 1]
```diff
@@ -815,87 +814,67 @@ async def _fetch_keys(
...
-                )
+                logger.warning("Error looking up keys from %s: %s", server_name, e)
             except Exception:
-                logger.exception("Error getting keys %s from %s", key_ids, server_name)
+                logger.exception("Error getting keys from %s", server_name)
 
         await yieldable_gather_results(get_key, keys_to_fetch)
         return results
 
     async def get_server_verify_key_v2_direct(
```
Comment (DMRobertson) [inline synapse/crypto/keyring.py:829]: Possibly this should be `get_server_verify_keyS_v2_direct` plural now?

File: synapse/crypto/keyring.py [hunk 2]
```diff
@@ -815,87 +814,67 @@ async def _fetch_keys(
 
         async def get_key(key_to_fetch_item: _FetchKeyRequest) -> None:
```
Comment (DMRobertson) [inline synapse/crypto/keyring.py:815]: Similarly should this be `get_keys` now?

File: changelog.d/14525.feature [hunk 3]
```diff
@@ -0,0 +1 @@
+Stop using deprecated `keyIds` parameter when calling `/_matrix/key/v2/server`.
```
Comment (clokep) [inline changelog.d/14525.feature:1]: Maybe combine the changelog for #14490 into this one?

File: synapse/crypto/keyring.py [hunk 4]
```diff
@@ -815,87 +814,67 @@ async def _fetch_keys(
...
             server_name = key_to_fetch_item.server_name
-            key_ids = key_to_fetch_item.key_ids
```
Comment (clokep) [inline synapse/crypto/keyring.py:818]: I did wonder if `key_ids` was even needed here now, but it seems to be used quite a bit.

**Comment by richvdh** [comment]
> Sanity check: do we expect all servers to be able to serve /key/v2/server without a key ID?

Synapse has ignored the keyID ever since `/key/v2/server` was implemented (https://github.com/matrix-org/synapse/pull/129).

File: synapse/crypto/keyring.py (hunk 1 above)
Comment (richvdh) [reply synapse/crypto/keyring.py:829]:
```suggestion
    async def get_server_verify_keys_v2_direct(
```

File: synapse/crypto/keyring.py [hunk 5]
```diff
@@ -815,87 +814,67 @@ async def _fetch_keys(
...
 
             try:
-                keys = await self.get_server_verify_key_v2_direct(server_name, key_ids)
+                keys = await self.get_server_verify_key_v2_direct(server_name)
```
Comment (richvdh) [inline synapse/crypto/keyring.py:819]:
```suggestion
                keys = await self.get_server_verify_keys_v2_direct(server_name)
```

File: synapse/crypto/keyring.py (hunk 1 above)
Comment (richvdh) [reply synapse/crypto/keyring.py:829]: it was just as capable of fetching multiple keys before, but sure

File: synapse/crypto/keyring.py (hunk 2 above)
Comment (richvdh) [reply synapse/crypto/keyring.py:815]:
```suggestion
        async def get_keys(key_to_fetch_item: _FetchKeyRequest) -> None:
```

File: synapse/crypto/keyring.py [hunk 6]
```diff
@@ -815,87 +814,67 @@ async def _fetch_keys(
...
             except KeyLookupError as e:
-                logger.warning(
-                    "Error looking up keys %s from %s: %s", key_ids, server_name, e
-                )
+                logger.warning("Error looking up keys from %s: %s", server_name, e)
             except Exception:
-                logger.exception("Error getting keys %s from %s", key_ids, server_name)
+                logger.exception("Error getting keys from %s", server_name)
 
         await yieldable_gather_results(get_key, keys_to_fetch)
```
Comment (richvdh) [inline synapse/crypto/keyring.py:826]:
```suggestion
        await yieldable_gather_results(get_keys, keys_to_fetch)
```

File: changelog.d/14525.feature (hunk 3 above)
Comment (richvdh) [reply changelog.d/14525.feature:1]: done

**Comment by richvdh** [comment]
(thanks for fixing my code)

**Pull request merged**